### ISSUE TITLE
[FR] Bug: Operator accuracy degradation on mthreads (google/gemma-3-1b-it)

### ISSUE THREAD

**Issue by Lxiparer** [FlagOpen/FlagGems#4820]
### Description

When running model inference on **google/gemma-3-1b-it** with FlagGems enabled on Moore Threads MTT S5000, GPQA Diamond accuracy drops from NV reference 27.0% to 12.0% (relative degradation 55.6%). This has been reproduced across two independent evaluation runs (14.89% and 12.0%), confirming it is a systematic issue rather than random variance.

### Environment

| Item | Value |
|------|-------|
| Hardware | mthreads |
| GPU | MTT S5000 × 8 |
| PyTorch | 2.7.1 |
| vLLM | 0.20.2 |
| FlagGems | 5.0.0 |
| Triton Backend | mtgpu |

### Steps to Reproduce

1. Set up environment on mthreads (MTT S5000)
2. Install dependencies (PyTorch 2.7.1 + Triton + FlagGems 5.0.0 + vLLM 0.20.2)
3. Start vLLM inference service with FlagGems enabled and `--enforce-eager`:
   ```bash
   MUSA_VISIBLE_DEVICES=2 VLLM_PLUGINS=fl USE_FLAGGEMS=1 \
     vllm serve google/gemma-3-1b-it \
       --host 0.0.0.0 --port 8000 \
       --enforce-eager \
       --max-model-len 8192 \
       --trust-remote-code
   ```
4. Run GPQA Diamond accuracy evaluation (50 questions, temperature=0, greedy decoding):
   ```bash
   python3 fast_gpqa.py --model gemma-3-1b-it --limit 50 --max-tokens 1024
   ```
5. Compare with NV reference score (27.0%)

### Expected Behavior

FlagGems operators should maintain accuracy within 5% relative degradation of the NV reference baseline.

### Actual Behavior

- NV reference (NVIDIA): **27.0%**
- FlagGems enabled (MUSA, first run): **14.89%**
- FlagGems enabled (MUSA, second run): **12.0%**
- Relative degradation: **55.56%** (threshold: 5%)

The issue is reproducible and consistent across runs, confirming a systematic precision problem on MUSA backend.

### Error Logs

```
/usr/local/lib/python3.10/dist-packages/triton/backends/mtgpu/compiler.py:391: RuntimeWarning: MUSA backend: num_stages>1 requested without using TME; falling back to num_stages=1.
  stages["ttgir"] = lambda src, metadata: self.make_ttgir(src, metadata, options, self.capability, self.warp_size)

[vllm_fl.dispatch.manager] Op 'attention_backend' using 'default.flagos' (kind=flagos, vendor=None)
[vllm_fl.dispatch.manager] Implementation 'default.flagos' failed for op 'attention_backend': TritonAttentionBackend requires CUDA but CUDA is not available. Falling back to vendor implementation.
[vllm_fl.dispatch.manager] Op 'attention_backend' fallback to 'vendor.musa' (kind=vendor, vendor=musa)
```

### FlagGems Integration Code

**gems.txt path**: `/root/gems.txt` (file not found or empty — plugin dispatch manages operator routing)

### Additional Context

- The issue appears to be specific to **mthreads** platform and may not reproduce on standard CUDA (NVIDIA) environments
- `--enforce-eager` is required because torch.compile crashes on MUSA (separate issue)
- The attention backend falls back from `default.flagos` (TritonAttentionBackend) to `vendor.musa` because CUDA is unavailable — this fallback path may contribute to precision differences
- Triton `num_stages` is forced to 1 on MUSA backend, which may affect kernel execution behavior

### Possible Directions

- Check the `vendor.musa` attention backend implementation for numerical precision (especially softmax FP32 accumulation)
- Verify Triton mtgpu kernel code generation preserves FP32 intermediate precision for reduction operations
- Run per-operator numerical comparison between MUSA Triton output and PyTorch eager reference output
- Investigate whether `num_stages=1` fallback affects reduction-type kernel precision
